PerfettoCds/Pipeline/Events/PerfettoSqlEvent.cs

```diff
@@ -0,0 +1,490 @@
+﻿using Microsoft.Performance.SDK;
+using Microsoft.Performance.SDK.Extensibility;
+using System;
+using Perfetto.Protos;
+
+namespace PerfettoCds.Pipeline.Events
+{
+    /// <summary>
+    /// Keep track of where we are in the varint, varfloat, and varstring arrays in a Perfetto batch result
+    /// </summary>
+    public class CellCounters
+    {
+        public int IntCounter { get; set; }
+        public int StringCounter { get; set; }
+        public int FloatCounter { get; set; }
+    }
+
+    /// <summary>
+    /// Base class for a row in a Perfetto SQL table
+    /// </summary>
+    public abstract class PerfettoSqlEvent : IKeyedDataType<String>
```

Comment:
This and the classes in this file make a decently large file. Consider breaking out to separate classes with a shared namespace?

Reply:
Separated out these classes in the new PerfettoProcessor lib